### PR TITLE
Adding support for "indices" routine

### PR DESCRIPTION
**Pull request by ipdemes**

### CODE REVIEW

File: cunumeric/module.py [hunk 1]
```diff
@@ -2203,6 +2203,68 @@ def where(a, x=None, y=None):
...
+    Parameters
+    ----------
+    dimensions : sequence of ints
+        The shape of the grid.
+    dtype : dtype, optional
+        Data type of the result.
+    sparse : boolean, optional
+        Return a sparse representation of the grid instead of a dense
+        representation. Default is False.
+        .. versionadded:: 1.17
```
Comment (manopapad) [inline cunumeric/module.py:2220]: Remove this

File: cunumeric/module.py [hunk 2]
```diff
@@ -2203,6 +2203,68 @@ def where(a, x=None, y=None):
 
 
 # Indexing-like operations
+def indices(dimensions, dtype=int, sparse=False):
```
Comment (manopapad) [inline cunumeric/module.py:2206]: Add a note here that the implementation is adapted from NumPy.

File: tests/integration/test_indices.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    assert np.array_equal(np_res, cn_res)
+
+    np_res = np.indices(dimensions, dtype=float)
+    cn_res = cn.indices(dimensions, dtype=float)
+    assert np.array_equal(np_res, cn_res)
+
+    np_res = np.indices(dimensions, sparse=True)
+    cn_res = cn.indices(dimensions, sparse=True)
+    print(np_res)
+    print(cn_res)
```
Comment (manopapad) [inline tests/integration/test_indices.py:40]: Remove this debug output

**Review by manopapad — APPROVED** [review]
Some minor comments, otherwise looks good

**Comment by ipdemes** [comment]
Thank you!

**Pull request merged**